ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/websocket/ClientSessions.kt

```diff
@@ -29,3 +33,55 @@ internal class DelegatingClientWebSocketSession(
     override val call: HttpClientCall,
     session: WebSocketSession
 ) : ClientWebSocketSession, WebSocketSession by session
+
+/**
+ * Serializes [data] to frame and enqueue this frame, may suspend if outgoing queue is full. May throw an exception if outgoing channel is already
+ * closed, so it is impossible to transfer any message. Frames that were sent after close frame could be silently
+ * ignored. Please note that close frame could be sent automatically in reply to a peer close frame unless it is
+ * raw websocket session.
+ */
```

Comment:
I suggest the following structure, with some minor fixes (articles, close frame -> Close frame, etc.):
> Serializes [data] to a frame and enqueues this frame. May suspend if the outgoing queue is full. If the outgoing channel is already closed, throws an exception, so it is impossible to transfer any message. Frames sent after a Close frame are silently ignored. Note that a Close frame could be sent automatically in reply to a peer's Close frame unless it is a raw WebSocket session.